Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -38,6 +38,23 @@ def relocate_dynamic_linkage(relocation)
     end
   end
 
+  # Ensures that an ad-hoc signature is present on all binaries,
+  # using the OS's builtin codesign tool.
+  # This is required on Apple Silicon on 11.0 or later, but not on Intel.
+  # While the build tools can generate valid ad-hoc signatures,
+  # Homebrew's mach_o tooling modifies the files without resigning.
+  # Consequently, we need to make sure we've applied a valid ad-hoc
+  # signature before running.
+  def codesign_mach_o_files
+    return if !Hardware::CPU.arm? || MacOS.version != :big_sur
```

Comment:
```suggestion
    return if !Hardware::CPU.arm? || MacOS.version < :big_sur
```

It doesn't change logic but better reflects a line "_This is required on Apple Silicon on 11.0 or later._" in my opinion.